internal/pkg/cli/init.go

```diff
@@ -153,22 +142,81 @@ func newInitOpts(vars initVars) (*initOpts, error) {
 		cmd:          command.New(),
 		sessProvider: sessProvider,
 	}
+	deployJobCmd := &deployJobOpts{
+		deployJobVars: deployJobVars{
+			envName:  defaultEnvironmentName,
+			imageTag: vars.imageTag,
+			appName:  vars.appName,
+		},
+		store:        ssm,
+		prompt:       prompt,
+		ws:           ws,
+		unmarshal:    manifest.UnmarshalWorkload,
+		sel:          sel,
+		spinner:      spin,
+		cmd:          command.New(),
+		sessProvider: sessProvider,
+	}
 
 	return &initOpts{
+		initVars:     vars,
 		ShouldDeploy: vars.shouldDeploy,
 
 		initAppCmd:   initAppCmd,
-		initSvcCmd:   initSvcCmd,
 		initEnvCmd:   initEnvCmd,
 		deploySvcCmd: deploySvcCmd,
+		deployJobCmd: deployJobCmd,
 
-		appName:        &initAppCmd.name,
-		svcType:        &initSvcCmd.wkldType,
-		svcName:        &initSvcCmd.name,
-		svcPort:        &initSvcCmd.port,
-		dockerfilePath: &initSvcCmd.dockerfilePath,
+		appName: &initAppCmd.name,
 
 		prompt: prompt,
+
+		setupWorkoadInit: func(o *initOpts, wkldType string) error {
+			wlInitializer := &initialize.WorkloadInitializer{Store: ssm, Ws: ws, Prog: spin, Deployer: deployer}
+			wkldVars := initWkldVars{
```

Comment:
huh i thought we had two different `Vars` structs one for Services and one for Jobs where they have embedded a common parent `initWorkloadVars` struct.

Allowing fields that are not used, like `port`, for job I think will lead to some confusion while working on these commands.